### ISSUE TITLE
Over training of model

### ISSUE THREAD

**Issue by gautamw7** [AvishakeAdhikary/Realtime-Sign-Language-Detection-Using-LSTM-Model#2]
Hey, I couldn't help but notice that you ran the model for 2000 ephochs, not an issue, but when you check the ehocs the accuracy reach 98 % and then dropped and then increased to 100% that, indicates that your model started to overfit, I wouldn't mind if you ran few test to check whether your model can run other words than food

1. Also you mentioned you that just run the notebook, but we can't because you have some dataset folder, which we don't have
2. Try to use someone else for your project demo, that would indicate that your model, can handle test cases too

Inform me, if your model works, I would love to collab with you, as I'm working on something simialr

**Comment by AvishakeAdhikary** [comment]
I appreciate the concern and to be very honest I am very happy that you want to collaborate with me, but I made this project during my masters' thesis, and a lot of time has passed and I am a Machine Learning Engineer now, working full time at a company and I don't have a lot of time to fulfill user requests based on their own presumed misconceptions.

Yes I made some small mistakes and yes there are a lot of new versions of dependencies that conflict the user code of that time that might not work on dependencies newer or older than the versions used.

But other than that I have explained about the dataset used in a similar [issue](https://github.com/AvishakeAdhikary/Realtime-Sign-Language-Detection-Using-LSTM-Model/issues/1).

Now coming to resolving your issues:
1. <i>`"Also you mentioned you that just run the notebook, but we can't because you have some dataset folder, which we don't have"`</i>
If you **properly** check the code, you will find a section:
```python
signs = ['cat','food','help']

parent_folder = 'MP_Data'

if not os.path.exists(parent_folder):
    os.mkdir(parent_folder)
    for sign in signs:
        sign_folder = os.path.join(parent_folder, sign)
        os.mkdir(sign_folder)
        for i in range(30):
            subfolder = os.path.join(sign_folder, str(i))
            os.mkdir(subfolder)
```
You see, that I check **if** a directory named `MP_Data` exists. **If it does not, I create one anyway using `os.mkdir()`, so it is inevitable that you and I will have the same directories for dataset training.**
2. <i>`"I wouldn't mind if you ran few test to check whether your model can run other words than food"`</i>
If you check the repository properly you will see that I **do** run the model more than once for the video which do have words other than "food" and the model worked absolutely amazing combining two different models to perfect a single task without creating much pressure on the system.
3. <i>`"Try to use someone else for your project demo, that would indicate that your model, can handle test cases too"`</i>
Once again, as you understand that I am already extremely busy with my corporate schedule and am unable to commit myself truly to the opensource community, and you are free to run the test cases yourself by forking the repo.

P.S. This is not a proper way to create and issue and order someone to do something and "notify" you for your own personal benefit. I sincerely believe that this is your first time and I am letting this one go and I also believe that you have a lot of scope for personal growth in this area.

Now let's move on to the motivational part. To be very honest, this repository has almost infinite scope for improvements. For example, I did not create a proper documentation about how to use the code, I did not explain much about how each and every line works, I did not consider all the types of cases where different types of dependencies of different versions would be used by different users and so on. And I have **absolutely no time** to contribute to the opensource community unless I am on holidays (which hardly happens). And the fact that you want to collaborate with me is great, but I cannot guarantee that I will be able to commit myself to this as much because I am a Machine Learning Engineer as well as a Machine Learning Lead now leading teams of people who have about 20+ years of experience who work under me, and the depth I know now is nothing compared to the knowledge I used to have before.

To sum up, if want to **contribute to this project** and make proper documentation and give test cases, you are more than open to do so. You just have to **fork** the repo, and **make changes to your own branch by creating one** and **creating a pull request**. If done properly, I will make sure there is nothing stopping you to contribute to the open source too. I believe I have made you motivated enough to **code and contribute**.

Happy Coding.
